rotkehlchen/tests/unit/test_globaldb.py

```diff
@@ -766,3 +797,95 @@ def test_add_user_owned_asset_nft(globaldb):
     new_assets = {x[0] for x in result}
     assert new_assets - initial_assets == {A_DAI.identifier, A_PICKLE.identifier, A_CRV.identifier}
     assert all(not x.startswith(NFT_DIRECTIVE) for x in new_assets)
+
+
+def test_asset_deletion(globaldb):
+    """This test checks that deletion of both evm token and normal asset works as expected"""
+    def check_tables(cursor, asset_id: str, expected_count: int, also_eth: bool):
```

Comment:
There really is no need to be passing the cursor here. You just make the test bigger. Just put the entire thing in a with cursor context and make the whole test smaller.